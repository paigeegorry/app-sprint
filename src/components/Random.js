import React from 'react';
import PropTypes from 'prop-types';

function Random({ random, onClick }) {
  return (
    <>
      <p>{random.body}</p>
      <button onClick={onClick}>Get Random</button>
    </>
  );
}

Random.propTypes = {
  random: PropTypes.object.isRequired,
  onClick: PropTypes.func.isRequired
};

export default Random;
