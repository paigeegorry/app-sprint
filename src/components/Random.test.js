import React from 'react';
import renderer from 'react-test-renderer';
import Random from './Random';

describe('Random snapshot', () => {
  it('can match a snapshot', () => {
    const tree = renderer.create(
      <Random
        random={{}}
        onClick={jest.fn()}
      />).toJSON();

    expect(tree).toMatchSnapshot();
  });
});
