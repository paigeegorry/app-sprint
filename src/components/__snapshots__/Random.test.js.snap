// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Random snapshot can match a snapshot 1`] = `
Array [
  <p />,
  <button
    onClick={[MockFunction]}
  >
    Get Random
  </button>,
]
`;
